freqtrade/freqai/data_kitchen.py

```diff
@@ -0,0 +1,1215 @@
+import copy
+import datetime
+import logging
+import shutil
+import sqlite3
+from pathlib import Path
+from typing import Any, Dict, List, Optional, Tuple
+
+import numpy as np
+import numpy.typing as npt
+import pandas as pd
+from pandas import DataFrame
+from sklearn import linear_model
+from sklearn.cluster import DBSCAN
+from sklearn.metrics.pairwise import pairwise_distances
+from sklearn.model_selection import train_test_split
+from sklearn.neighbors import NearestNeighbors
+
+from freqtrade.configuration import TimeRange
+from freqtrade.data.history.history_utils import refresh_backtest_ohlcv_data
+from freqtrade.exceptions import OperationalException
+from freqtrade.resolvers import ExchangeResolver
+from freqtrade.strategy.interface import IStrategy
+
+
+SECONDS_IN_DAY = 86400
+SECONDS_IN_HOUR = 3600
+
+logger = logging.getLogger(__name__)
+
+
+class FreqaiDataKitchen:
+    """
+    Class designed to analyze data for a single pair. Employed by the IFreqaiModel class.
+    Functionalities include holding, saving, loading, and analyzing the data.
+
+    Record of contribution:
+    FreqAI was developed by a group of individuals who all contributed specific skillsets to the
+    project.
+
+    Conception and software development:
+    Robert Caulk @robcaulk
+
+    Theoretical brainstorming:
+    Elin Törnquist @th0rntwig
+
+    Code review, software architecture brainstorming:
+    @xmatthias
+
+    Beta testing and bug reporting:
+    @bloodhunter4rc, Salah Lamkadem @ikonx, @ken11o2, @longyu, @paranoidandy, @smidelis, @smarm
+    Juha Nykänen @suikula, Wagner Costa @wagnercosta
+    """
+
+    def __init__(
+        self,
+        config: Dict[str, Any],
+        live: bool = False,
+        pair: str = "",
+    ):
+        self.data: Dict[Any, Any] = {}
+        self.data_dictionary: Dict[Any, Any] = {}
```

Comment:
Is there a way to better type this?

for example via `TypedDict` [docs](https://docs.python.org/3/library/typing.html#typing.TypedDict) - that'd enable mypy (and also editors) to properly help us detect bugs ... 
I know we're not using this elsewhere - but i think it would help us a lot in this case - where "any/any" is used to type the dictionary...